app/views/courses/CoursesView.coffee

```diff
@@ -91,10 +92,16 @@ module.exports = class CoursesView extends RootView
       @renderSelectors '#join-class-form'
       return
     @renderSelectors '#join-class-form'
-    newClassroom = new Classroom()
-    newClassroom.joinWithCode(@classCode)
-    newClassroom.on 'sync', @onJoinClassroomSuccess, @
-    newClassroom.on 'error', @onJoinClassroomError, @
+    if me.get('emailVerified')
+      newClassroom = new Classroom()
+      jqxhr = newClassroom.joinWithCode(@classCode)
+      @listenTo newClassroom, 'join:success', -> @onJoinClassroomSuccess(newClassroom)
+      @listenTo newClassroom, 'join:error', -> @onJoinClassroomError(newClassroom, jqxhr)
+    else
+      modal = new JoinClassModal({ @classCode })
+      @openModalView modal
+      @listenTo modal, 'join:success', @onJoinClassroomSuccess
+      @listenTo modal, 'join:error', @onJoinClassroomError
```

Comment:
Hmm, I wonder if Promises would be a better way to handle this. The jqxhr already is one, and perhaps the JoinClassModal could provide one. Then this view could hook into whichever Promise is produced. When you're doing async work and the only events that can happen are success or failure, Promises provide a nice structure, and JS seems to be headed toward their dominance. Just food for thought, let's leave this as is for now.